regression_test.go

```diff
@@ -72,5 +73,79 @@ func Test_Regression_29324(t *testing.T) {
 	iter = table.List(txn, idIndex.Query("a"))
 	items = Collect(iter)
 	assert.Len(t, items, 0, "Get(\"a\") should return nothing")
+}
+
+// The watch channel returned by Changes() must be a closed one if there
+// is anything left to iterate over. Otherwise on partial iteration we'll
+// wait on a watch channel that reflects the changes of a full iteration
+// and we might be stuck waiting even when there's unprocessed changes.
+func Test_Regression_Changes_Watch(t *testing.T) {
+	db, table, _ := newTestDB(t)
+
+	wtxn := db.WriteTxn(table)
+	changes, err := table.Changes(wtxn)
+	require.NoError(t, err, "Changes")
+	wtxn.Commit()
+
+	n := 0
+	_, _, ok := changes.Next()
+	require.False(t, ok, "expected no changes")
+
+	// On first call to Watch() the iterator is refreshed and a closed watch channel
+	// is returned.
```

Comment:
If I get this right, this comment is not true anymore now that the initial watch channel returned is not a `closedWatchChannel`. Should we update this in the last commit to avoid confusion?

Reply:
It is still true. `Watch` returns a closed watch channel when the iterator has not been consumed.